### DISCUSSION TITLE
Question: Detecting keys/properties not in schema

### DISCUSSION THREAD

**Discussion by lmartinking** [santhosh-tekuri/jsonschema#188] [Q&A]
Hi, I was wondering if it's possible to detect properties/keys not defined in the schema?

Thanks in advance!

**Comment by santhosh-tekuri** [comment]
use [additionalProperties](https://json-schema.org/understanding-json-schema/reference/object#additionalproperties) keyword with `false` value

```
"additionalProperties": false
```

**Comment by jantari** [comment]
Hello, `"additionalProperties": false` is exactly the behavior I want.

However I have to add it to every object in the whole schema which is tedious and error-prone. It would be much nicer if there was an option for globally strict validation by default, e.g. assume `"additionalProperties": false` for any object in the schema where "additionalProperties" is not explicitly set to something else.
